onnxruntime/core/providers/openvino/backends/basic_backend.cc

```diff
@@ -529,74 +368,99 @@ void BasicBackend::Infer(OrtKernelContext* ctx) {
+
+      ORT_ENFORCE(ov_tensor->get_byte_size() == ort_tensor.GetTensorSizeInBytes(),
+                  log_tag + "Output tensor size mismatch for " + output_info.name);
+
+      std::memcpy(ort_tensor.GetTensorMutableRawData(),
+                  ov_tensor->data(),
+                  ov_tensor->get_byte_size());
     }
-    try {
-      CompleteAsyncInference(context, infer_request);
-    } catch (const std::runtime_error& e) {
-      // If the inference fails (exception from ov::InferRequest::infer()),
-      // we need to put the infer_request back into the pool to avoid deadlocks
-      // and to allow the next inference request to proceed.
-      inferRequestsQueue_->putIdleRequest(std::move(infer_request));
-      ORT_THROW(log_tag + " Exception at CompleteAsyncInference: " + e.what());
+  } else {
+    // Static shape inference
+
+    // Bind inputs
+    for (const auto& input_info : bindings_->network_inputs_) {
+      infer_request->SetTensor(input_info, const_cast<void*>(context.GetInput(input_info.onnx_index).GetTensorRawData()));
     }
 
-    // Get Output tensors
-    LOGS_DEFAULT(INFO) << log_tag << "Inference successful";
-    // Enable CI Logs
-    if (IsCILogEnabled()) {
-      std::cout << "Inference successful" << std::endl;
+    // Bind outputs
+    for (const auto& output_info : bindings_->network_outputs_) {
+      infer_request->SetTensor(output_info, context.GetOutput(output_info.onnx_index, output_info.shape).GetTensorMutableRawData());
     }
 
-    // Create a duplicate infer_request_ shared ptr on the stack in the current local scope,
-    // as the infer_request gets freed in the next stage the reference count for the infer_request decrements &
-    // thus we dont have any dangling ptr leading to seg faults in the debug mode subsequent execution call
-    OVInferRequestPtr infer_request_ = infer_request;
+    // Run Inference
+    infer_request->Infer();
+  }
 
-    // Once the inference is completed, the infer_request becomes free and is placed back into pool of infer_requests_
-    inferRequestsQueue_->putIdleRequest(std::move(infer_request));
-#ifndef NDEBUG
-    if (openvino_ep::backend_utils::IsDebugEnabled()) {
-      inferRequestsQueue_->printstatus();  // Printing the elements of infer_requests_ vector pool only in debug mode
-      std::string& hw_target = session_context_.device_type;
-      printPerformanceCounts(std::move(infer_request_), std::cout, hw_target);
+    // Fill constant outputs if needed
+    for (const auto& [name, node] : const_outputs_map_) {
+        Ort::UnownedValue output_tensor = GetOutputTensor(context,
+                                                        name,
+                                                        subgraph_context_.output_names,
+                                                        node);
+        auto mem_info = output_tensor.GetTensorMemoryInfo();
+        ORT_ENFORCE(mem_info.GetAllocatorName() != OpenVINO_GPU,
```

Comment:
We can remove this log as IO Buffering is not supported anyway and we will put a depracation notice for context